app/src/main/java/com/example/cs4520_twitter/data_layer/database/UserProfileDAO.kt

```diff
@@ -0,0 +1,20 @@
+package com.example.cs4520_twitter.data_layer.database
+
+import androidx.room.Dao
+import androidx.room.Insert
+import androidx.room.Query
+
+/**
+ * User profile DAO
+ */
+@Dao
+interface UserProfileDao {
+    @Insert
+    suspend fun insert(userProfile: UserProfileEntity)
+
+    /**
```

Comment:
Do we want to implement an update function for making edits to a user's profile/following?

Reply:
sure so would this be kinda of like accept multiple various args or do we want like an updateDescription, updateUsername esque style methods?

Reply:
@tomgrbz I think separate would be more organized, especially since the SQL query requires the field you want to update to be specified. Also, going back to the Figma wireframe, there's a save icon next to each input bar on the Edit Profile screen, so we'll be updating individual fields as opposed to the whole profile in one go.